packages/@aws-cdk/aws-stepfunctions/lib/states/task-base.ts

```diff
@@ -0,0 +1,311 @@
+    return this.taskMetric(this.taskMetrics().metricPrefixSingular, 'RunTime', { statistic: 'avg', ...props });
+  }
+
+  /**
+   * The interval, in milliseconds, for which the activity stays in the schedule state.
+   *
+   * @default average over 5 minutes
+   */
+  public metricScheduleTime(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixSingular, 'ScheduleTime', { statistic: 'avg', ...props });
+  }
+
+  /**
+   * The interval, in milliseconds, between the time the activity is scheduled and the time it closes.
+   *
+   * @default average over 5 minutes
+   */
+  public metricTime(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixSingular, 'Time', { statistic: 'avg', ...props });
+  }
+
+  /**
+   * Metric for the number of times this activity is scheduled
+   *
+   * @default sum over 5 minutes
+   */
+  public metricScheduled(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixPlural, 'Scheduled', props);
+  }
+
+  /**
+   * Metric for the number of times this activity times out
+   *
+   * @default sum over 5 minutes
+   */
+  public metricTimedOut(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixPlural, 'TimedOut', props);
+  }
+
+  /**
+   * Metric for the number of times this activity is started
+   *
+   * @default sum over 5 minutes
+   */
+  public metricStarted(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixPlural, 'Started', props);
+  }
+
+  /**
+   * Metric for the number of times this activity succeeds
+   *
+   * @default sum over 5 minutes
+   */
+  public metricSucceeded(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixPlural, 'Succeeded', props);
+  }
+
+  /**
+   * Metric for the number of times this activity fails
+   *
+   * @default sum over 5 minutes
+   */
+  public metricFailed(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixPlural, 'Failed', props);
+  }
+
+  /**
+   * Metric for the number of times the heartbeat times out for this activity
+   *
+   * @default sum over 5 minutes
+   */
+  public metricHeartbeatTimedOut(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.taskMetric(this.taskMetrics().metricPrefixPlural, 'HeartbeatTimedOut', props);
+  }
+
+  protected whenBoundToGraph(graph: StateGraph) {
+    super.whenBoundToGraph(graph);
+    for (const policyStatement of this.taskPolicies() || []) {
+      graph.registerPolicyStatement(policyStatement);
+    }
+  }
+
+  protected abstract taskMetrics(): TaskMetricsConfig;
+
+  protected abstract taskPolicies(): iam.PolicyStatement[];
+
+  protected abstract renderTask(): any;
```

Comment:
Make this a property instead of a function.

Reply:
offline we decided to leave this as it gives concrete classes the ability to supplement any properties or additional fields. An alternative was for concrete classes to directly override `toStateJson` and provide all the missing information. However, that's a public method and we lose out on being able to guide that task state is rendered somewhere by dropping this method.